pkgs/tools/text/gawk/gawkextlib.nix

```diff
@@ -0,0 +1,97 @@
+{ stdenv, recurseIntoAttrs, fetchgit, writeText, pkgconfig, autoreconfHook, autoconf, automake
+, libtool, texinfo, gettext, gawk, rapidjson, gd, shapelib, libharu, lmdb, gmp
+, glibcLocales, mpfr, more, postgresql, hiredis, expat, tre, makeWrapper
+}:
+
+let
+  buildGawkextlibExtension = extension: extraDeps:
+    stdenv.mkDerivation rec {
+      pname = "gawkextlib-${extension}";
+      version = "unstable-2019-11-21";
+
+      src = fetchgit {
+        url = "git://git.code.sf.net/p/gawkextlib/code";
+        rev = "f70f10da2804e4fd0a0bac57736e9c1cf21e345d";
+        sha256 = "0r8fz89n3l4dfszs1980yqj0ah95430lj0y1lb7blfkwxa6c2xik";
+      };
+
+      nativeBuildInputs =
+        [ autoconf automake libtool autoreconfHook pkgconfig texinfo gettext ];
+
+      buildInputs = [ gawk gawkextlib ] ++ extraDeps;
+
+      configureFlags = [
+        "--with-gawkextlib=${gawkextlib}/lib/"
+        "LDFLAGS=-L${gawkextlib}/lib"
+      ];
+
+      postPatch = ''
+        cd ${extension}
+      '';
+      installPhase = ''
+        runHook preInstall
+        mkdir -p $out/lib
+        cp ./.libs/* $out/lib
+        runHook postInstall
+      '';
+
+      # Needed for some tests
+      preCheck = ''
+        AWKLIBPATH=${gawk}/lib/gawk
```

Comment:
Shouldn't setupHook of `gawkextlib` handle this?

Reply:
No. gawk itself should. This causes a problem when trying to do `nix-shell --pure -p gawk -p gawkextlib.json` i think gawk does some relative path searching when AWKLIBPATH is not defined, but as soon as you start defining it (and the tests in the select extension do), you need that in the path. 

Not sure if I should just add a setupHook to gawk itself.

Reply:
Added the `${gawk}/lib/gawk` piece to a the gawkextlib setupHook only. This means it is inherited by the actual extensions and enables the use of
```
nix-shell -I nixpkgs=. --pure -p gawk -p gawkextlib.json
```
without breaking things like time.so and filefuncs.so (these come with gawk itself).

This made a lot of other things much simpler at the cost of two separate setupHooks.

Reply:
Maybe you can use gawklibpath_append? (should be available after you source @gawk@/etc/profile.d/gawk.sh, need to set `gawk` attribute for @gawk@ to work)
It looks like it will not work with set -u, though. You might need to do
```
    local oldOpts="-u"
    shopt -qo nounset || oldOpts="+u"
    set +u

    set "$oldOpts"
```
or just copy paste their implementation into our setupHook.